test/config_test/config_test.cc

```diff
@@ -92,8 +92,13 @@ uint32_t run(const std::string& directory) {
   RELEASE_ASSERT(::chdir(directory.c_str()) == 0);
   uint32_t num_tested = 0;
   for (const std::string& filename : TestUtility::listFiles(directory, false)) {
-    ConfigTest config(filename);
+    Server::TestOptionsImpl options(filename, Network::Address::IpVersion::v6);
+    ConfigTest test1(options);
     num_tested++;
+    if (StringUtil::endsWith(filename, ".json")) {
+      ConfigTest test2(*options.asConfigJson());
```

Comment:
This is your leak. You never free the return value of asConfigJson(), and nothing owns it.

Reply:
Arf; thank you. Fixed.